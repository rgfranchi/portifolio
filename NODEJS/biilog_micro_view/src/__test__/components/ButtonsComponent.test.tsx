import React from "react";

import Enzime, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import {
  ButtonIconUpdate,
  ButtonIconDelete,
  ButtonCreate,
  ButtonsPanel,
  IButtonsPanel,
} from "../../components/ButtonsComponent";

Enzime.configure({ adapter: new Adapter() });

const linkTo: string = "http://linkteste.com.br";

describe("ButtonsComponent snapshot", () => {
  test("ButtonIconUpdate", () => {
    const wrapper = shallow(<ButtonIconUpdate linkTo={linkTo} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("ButtonIconDelete", () => {
    const wrapper = shallow(<ButtonIconDelete execute={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("ButtonCreate", () => {
    const wrapper = shallow(
      <ButtonCreate
        linkTo={linkTo}
        text="Novo Item"
        variant="outline-success"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("ButtonsPanel", () => {
    const btList: IButtonsPanel = {
      column: 4,
      list: [
        {
          id: 1,
          text: "Bt Test 1",
          execute: () => {},
          variant: "outline-primary",
        },
        {
          id: 2,
          text: "Bt Test 2",
          execute: () => {},
          variant: "outline-primary",
        },
        {
          id: 3,
          text: "Bt Test 3",
          execute: () => {},
          variant: "outline-primary",
        },
      ],
    };
    const wrapper = shallow(
      <ButtonsPanel column={btList.column} list={btList.list} />
    );
    const button = wrapper.find("Button");
    expect(button.length).toBe(btList.list.length);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("ButtonsComponent execute", () => {
  test("ButtonIconDelete", () => {
    const clickEd = jest.fn();
    const button = shallow(<ButtonIconDelete execute={clickEd} />);
    button.find("Button").simulate("click");
    expect(clickEd.mock.calls.length).toEqual(1);
  });
});
