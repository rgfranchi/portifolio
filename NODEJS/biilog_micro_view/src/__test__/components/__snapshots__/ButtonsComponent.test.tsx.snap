// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsComponent snapshot ButtonCreate 1`] = `
<Button
  active={false}
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "Link",
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  disabled={false}
  to="http://linkteste.com.br"
  variant="outline-success"
>
  Novo Item
</Button>
`;

exports[`ButtonsComponent snapshot ButtonIconDelete 1`] = `
<Button
  active={false}
  disabled={false}
  onClick={[Function]}
  size="sm"
  variant="danger"
>
  <AiFillDelete />
</Button>
`;

exports[`ButtonsComponent snapshot ButtonIconUpdate 1`] = `
<Button
  active={false}
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "Link",
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  disabled={false}
  size="sm"
  to="http://linkteste.com.br"
  variant="primary"
>
  <FiEdit />
</Button>
`;

exports[`ButtonsComponent snapshot ButtonsPanel 1`] = `
<Fragment>
  <Row
    key="0"
    noGutters={false}
  >
    <Col
      key="0"
    >
      <Button
        active={false}
        block={true}
        disabled={false}
        id={1}
        key="0"
        onClick={[Function]}
        variant="outline-primary"
      >
        Bt Test 1
      </Button>
    </Col>
    <Col
      key="1"
    >
      <Button
        active={false}
        block={true}
        disabled={false}
        id={2}
        key="1"
        onClick={[Function]}
        variant="outline-primary"
      >
        Bt Test 2
      </Button>
    </Col>
    <Col
      key="2"
    >
      <Button
        active={false}
        block={true}
        disabled={false}
        id={3}
        key="2"
        onClick={[Function]}
        variant="outline-primary"
      >
        Bt Test 3
      </Button>
    </Col>
  </Row>
</Fragment>
`;
